zerver/lib/actions.py

```diff
@@ -1362,16 +1362,26 @@ def do_deactivate_stream(
     new_name = (hashed_stream_id + "!DEACTIVATED:" + old_name)[: Stream.MAX_NAME_LENGTH]
 
     stream.name = new_name[: Stream.MAX_NAME_LENGTH]
-    stream.save(update_fields=["name", "deactivated", "invite_only"])
+    with transaction.atomic():
```

Comment:
This ends up nesting `transaction.atomic`, which I think we determined was a bad idea without using `savepoint=False` or `durable=True` on the inner calls, right?

Reply:
Added arguments to these `atomic()` instances, per the discussion that happened in #backend since